helix-core/src/main/java/org/apache/helix/manager/zk/CallbackHandler.java

```diff
@@ -721,6 +714,21 @@ void reset(boolean isShutdown) {
     }
   }
 
+  /**
+   * The package private method only closes the batchCallbackProcessor thread
+   * *Caution*: currently it's only used during disconnecting from ZK and the
+   * listeners unsubscription will be taken care by zkClient directly
+   */
+  void closeBatchCallbackProcessor() {
```

Comment:
Besides my concern of not sending FINALIZE, this method is a subset of void reset(boolean isShutdown); I think you can avoid duplicate code.

Of course, please address my concern about not having the FINALIZE first. Then we can revisit this one.